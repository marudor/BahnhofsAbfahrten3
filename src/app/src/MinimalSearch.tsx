import { AbfahrtenResult } from 'types/iris';
import { Button, ScrollView, Text, TextInput } from 'react-native';
import Axios from 'axios';
import React, { useCallback, useState } from 'react';
import useStorage from 'shared/hooks/useStorage';

Axios.defaults.baseURL = 'https://marudor.de';

let cancelGetAbfahrten = () => {};

export const fetchAbfahrten = async (
  stationId: string,
  lookahead: string,
  lookbehind: string
): Promise<AbfahrtenResult> => {
  cancelGetAbfahrten();
  const r = await Axios.get<AbfahrtenResult>(
    `/api/iris/v1/abfahrten/${stationId}`,
    {
      cancelToken: new Axios.CancelToken(c => {
        cancelGetAbfahrten = c;
      }),
      params: {
        lookahead,
        lookbehind,
      },
    }
  );

  return r.data;
};

const MinimalSearch = () => {
  const storage = useStorage();

  const [searchString, setSearchString] = useState<string>(
    storage.get('test') ?? ''
  );
  const [abfahrten, setAbfahrten] = useState<AbfahrtenResult | undefined>();
  const fetch = useCallback(() => {
    fetchAbfahrten(searchString, '150', '0').then(setAbfahrten);
    storage.set('test', searchString);
  }, [searchString, storage]);

  return (
    <ScrollView>
      <TextInput onChangeText={setSearchString} value={searchString} />
      <Button title="Search" onPress={fetch} />
      {abfahrten &&
        abfahrten.departures.map(a => (
          <Text key={a.rawId}>
            {a.train.name} -&gt; {a.destination}
          </Text>
        ))}
    </ScrollView>
  );
};

export default MinimalSearch;
