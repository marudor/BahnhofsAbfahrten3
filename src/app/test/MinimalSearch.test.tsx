import { render } from './TestHelper';
import MinimalSearch from '@/MinimalSearch';
import React from 'react';

describe('Simple Proof of Concept Test', () => {
  it('MinimalSearch renders', () => {
    const { baseElement } = render(<MinimalSearch />);

    expect(baseElement).toMatchSnapshot();
  });
});
