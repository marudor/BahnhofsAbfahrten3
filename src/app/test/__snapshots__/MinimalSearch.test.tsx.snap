// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Proof of Concept Test MinimalSearch renders 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <ScrollView>
      <TextInput
        allowFontScaling={true}
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
        value=""
      />
      <Button
        title="Search"
      />
    </ScrollView>
  </View>
</View>
`;
